portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,35 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+    
+  
+ 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Leah!");
+    ArrayList<String> jsonData = new ArrayList<String>();  
+    jsonData.add("nice!");
+    jsonData.add("cool!");
+    jsonData.add("awesome!");
+    convertToJsonUsingGson(jsonData);
+    response.setContentType("application/json;");
+    response.getWriter().println(jsonData);
+  }
+
+  private String convertToJsonUsingGson(ArrayList<String> jsonData) {
+    Gson gson = new Gson();
```

Comment:
For instances like this that don't maintain state and might cost a lot in terms of performance, it's beneficial to keep a reference to it as a field in the class. That way you can reuse the instance instead of recreating it every time.

Reply:
Ok, will do. Can you clarify what you mean by "don't maintain state"?

Reply:
What I mean by state is something that changes. Gson doesn't really keep track of anything, so every call to toJson with the same inputs will always yield the same result. Something that has state might change behavior based on what that state is. In that case, you might be wary of keeping it as a field in the class, opting maybe to create it newly every time. But it all depends on your use case. Sometimes you're relying on that state and want it to do something different depending on what state it's in. 

In this case, you do want it to do the same thing every time, and the same instance of gson will, so it's OK to have a persistent instance of it as a field in the class.

Reply:
That's a great explanation, I understand now. Thanks!